### ISSUE TITLE
Ability to airdrop/sharedrop/pay dividends of one asset to all holders of another.

### ISSUE THREAD

**Issue by clockworkgr** [bitshares/bitshares-core#1172]
As a `user/asset issuer` I want `to have a transfer_asset_to_asset_holders API call` so that `I can perform airdrops/sharedrops or dividend payouts`.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [x ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [x] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [x] Performance (system or user efficiency, etc.)
- [x] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [x] UX (the User Experience)
- [ ] Other (please add below)

**Additional Context (optional)**

Case 1: I own asset COOLASSET and decide to airdrop 50% of the total 1000000 COOLASSET supply to holders of BTS. I issue the relevant amount to myself and use transfer_asset_to_asset_holders(500000, 'COOLASSET','BTS')

Case 2: As a gateway operator, I collect market fees on all assets I provide gateway services for. Once a month, I want to share 50% of all market fee profits to holders of my GATEWAYCOIN. So for each gatewy asset, I withdraw market fees, calculate 50% profit share (e.g. say 500 GATEWAY.UIA1) and use transfer_asset_to_asset_holders(500,'GATEWAY.UIA1','GATEWAYCOIN')

I believe PBSA has already implemented most of the code needed for this in peerplays.

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by clockworkgr** [comment]
As a side note, the same code could be called upon to allow airdrops on asset creation time.

So from a UX point of view, the user would set all his UIA parameters just like now and on the supply settings, once a max supply is entered, he can choose a "percentage_to_airdrop" (e.g. 100%), an "airdrop_target_asset" (e.g. BTS) and a future block no for the snapshot (e.g. block 30000000).

Thus creating the asset and planning the airdrop at the same time.

**Comment by cogutvalera** [comment]
good idea ! nice !

**Comment by abitmore** [comment]
Interesting. I think there is a BSIP already.

Do it at maintenance interval? Deduct (transfer/issue) fees relative to number of holders?

**Comment by clockworkgr** [comment]
Yep...a superset of the functionality is mentioned here: https://github.com/bitshares/bsips/blob/master/bsip-0020.md

So change this issue to let's get BSIP-20 done instead :)

Maintenance interval sounds good.. Doesn't seem like an operation that will ever be required to be performed instantly.

Fees based on no of holders sounds good. Although we should probably make a special allowance for/treat core token BTS differently.

**Comment by clockworkgr** [comment]
@ryanRfox you were part of the original conversation so I'm tagging you here... As you can see there already was a BSIP...I was unaware

**Comment by abitmore** [comment]
Please be aware that balance look-up impacts performance, as mentioned in https://github.com/bitshares/bitshares-core/issues/1083#issuecomment-399992509, utilizing more than 7% of total replay time. Balance-modify may impact performance (much) further, as we need to insert 100k objects to object database in a few seconds.

**Comment by clockworkgr** [comment]
Hence why I checked performance when submitting the issue :)

I really don't know what the impact would be. I hope it's something that can be handled. It can be a high-fee operation if needed, as long as it's possible.

**Comment by clockworkgr** [comment]
https://github.com/BunkerChainLabsInc/peerplays-profitshare

**Comment by ryanRfox** [comment]
Currently BSIP 20 (UIA) and BSIP 19 (MPA) are `Deferred`. @xeroc may I request you to remind us of the conclusion at that time?

**Comment by clockworkgr** [comment]
Bumping this. @xeroc  / @ryanRfox ?

Had this request in UI too: https://github.com/bitshares/bitshares-ui/issues/2458

**Comment by pmconrad** [comment]
Related: https://github.com/bitshares/bsips/pull/132
